Description: Allow generated_files.sh to be used before building
Author: Faidon Liambotis <paravoid@debian.org>
Last-Update: 2014-03-05
 generated_files.sh barfs when the hhvm binary is not present. However, the
 actual binary is only needed for running the license target and not the other
 (parser) targets. Move the stanza inside the license block, thus allowing the
 script to be used to build (flex/bison) the parsers before actually building
 anything.

--- a/hphp/tools/generated_files.sh
+++ b/hphp/tools/generated_files.sh
@@ -15,16 +15,6 @@ cd $HPHP_HOME
 
 VERBOSE=1
 
-HHVM=$HPHP_HOME/hphp/hhvm/hhvm
-if [ -x "$HHVM" ]; then
-  export HHVM_SYSTEMLIB=$HPHP_HOME/bin/systemlib.php
-else
-  HHVM=`which hhvm`
-  export HHVM_SYSTEMLIB=`dirname $HHVM`/systemlib.php
-fi
-
-[ ! -x "$HHVM" ] && check_err 1 "$HHVM is not executable"
-
 HPHP_TOOLS=$HPHP_HOME/hphp/tools/
 
 if [ "$1" = "help" ]; then
@@ -82,6 +72,16 @@ if [ "$1" = "ini-parser" -o "$1" = "all" ]; then
 fi
 
 if [ "$1" = "license" -o "$1" = "all" ]; then
+  HHVM=$HPHP_HOME/hphp/hhvm/hhvm
+  if [ -x "$HHVM" ]; then
+    export HHVM_SYSTEMLIB=$HPHP_HOME/bin/systemlib.php
+  else
+    HHVM=`which hhvm`
+    export HHVM_SYSTEMLIB=`dirname $HHVM`/systemlib.php
+  fi
+
+  [ ! -x "$HHVM" ] && check_err 1 "$HHVM is not executable"
+
   [ $VERBOSE -eq 1 ] && echo "Updating license headers"
   # TODO: At the moment, license.php fails on PCRE_ERROR_MATCHLIMIT
   # Fix that script then change this to detect errors properly
